01-import_and_filter.py

```diff
@@ -91,12 +105,30 @@ def run_filter(subject, run=None, session=None):
 
         raw.resample(config.resample_sfreq, npad='auto')
 
-    # Prepare a name to save the data
-    fpath_out = op.join(config.bids_root, 'derivatives', subject_path)
+    # Prepare the pipeline directory in /derivatives
+    fpath_out = op.join(config.bids_root, 'derivatives',
+                        'mne-study-template', subject_path)
     if not op.exists(fpath_out):
         os.makedirs(fpath_out)
-    fname_out = op.join(fpath_out, bids_basename + '_filt_raw.fif')
 
+        # Write a dataset_description.json for the pipeline
+        ds_json = dict()
+        ds_json['Name'] = 'mne-study-template outputs'
+        ds_json['BIDSVersion'] = BIDS_VERSION
+        ds_json['PipelineDescription'] = {
+            'Name': 'mne-study-template',
+            'Version': config.VERSION,
+            'CodeURL': config.CODE_URL,
+            }
+        ds_json['SourceDatasets'] = {
+            'URL': _provide_testing_data().get(config.bids_root, 'n/a'),
```

Comment:
hmmm ... this is hard coded? what happens for user datasets?

Reply:
User datasets get the "'n/a" label, and they can fill it themselves. We can only automate so much, similar to what we do with `dataset_description` in mne-bids

Reply:
Fair enough